File: mocha/test/checkout.spec.js
const { expect } = require('chai')
const { isFirstAttempt } = require('./attempt')

// Mocha's spec reporter numbers its failures and prints the suite path across
// several indented lines rather than on one, which is a shape of its own:
//
//   1) checkout
//        should apply a discount code before tax:
//      AssertionError: expected 4200 to equal 3800
describe('checkout', function () {
  it('totals the basket', function () {
    expect(1 + 1).to.equal(2)
  })

  it('applies a discount code before tax, and rounds half up', function () {
    if (isFirstAttempt()) {
      expect(4200).to.equal(3800)
    }
  })
})

File: rspec/spec/checkout_spec.rb
require_relative 'spec_helper'

# RSpec joins the describe path with spaces rather than a separator, and prints
# the failure under a numbered heading with the whole path on one line.
RSpec.describe 'Checkout' do
  it 'totals the basket' do
    expect(1200 + 800).to eq(2000)
  end

  it 'applies a discount code before tax, and rounds half up' do
    expect(4200).to eq(3800) if first_attempt?
  end
end
